compiler/src/main/java/com/squareup/anvil/compiler/ModuleMergerIr.kt

```diff
@@ -291,12 +287,11 @@ internal class ModuleMergerIr(
 
           putValueArgument(
             index = 1,
-            valueArgument = IrVarargImpl(
-              startOffset = 0,
-              endOffset = 0,
-              type = varargArguments[0].type,
-              varargElementType = varargArguments[0].varargElementType,
-              elements = varargArguments.flatMap { it.elements }
+            valueArgument = irVararg(
+              elementType = varargArguments[0].varargElementType,
+              // These are always IrExpression instances too
+              values = varargArguments.flatMap { it.elements }
+                .filterIsInstance<IrExpression>()
```

Comment:
Are we sure this is guaranteed? Wondering why the extension function enforces `IrExpressions` when the impl class doesn't

Reply:
Yeah if you look at the impl that's all it allows. My guess is it's a bit of a marker interface tomfoolery